### PR TITLE
Update the contributor ladder.

### LINKED ISSUE #390: Enhance the contributor ladder to better attract and motivate developers.
This enhancement aim to redesign our contributor ladder to better attract new contributors and motivate them.

Here list issue we found, and put them in different stage:

## Under discussion

- Lower the bar for each contributor level
- Introduce a new role for inactive maintainers
- Eliminate the reviewer role since it is not being used; consider adding a "lead" role for each area
- Consider treating addons as a special interest group.

## TODO

- 

## InProgress

- 

## Review

- 

Anyone can put your thought as comments under this issue, the description of the issue will capture the overall progress.

### PR DESCRIPTION
**Pull request by xuezhaojun**
Fixes: https://github.com/open-cluster-management-io/ocm/issues/390

### CODE REVIEW

File: CONTRIBUTOR_LADDER.md [hunk 1]
```diff
@@ -19,6 +19,13 @@ Each of the contributor roles below is organized into lists of three types of th
 
 Open Cluster Management (OCM) is a project made up of several Subprojects, such as Foundation and Policy.  This contributor ladder applies to each Subproject.  As such, a contributor advances in each Subproject separately, and may be a Member in one Subproject while being a Maintainer in another.  For the Open Cluster Management project overall, the contributor is considered to be the highest level they have reached in any Subproject.  Thus, someone who is a Project Member in one Subproject is a Project Member of OCM overall.
 
+| Role | Responsibilities | Requirements | Privileges |
+| ---- | ---------------- | ------------ | ---------- |
+| [Contributor](#contributor) | Casual contributor to the project | n/a | Invitations to contributor events, Eligible to become an Organization Member |
+| [Organization Member](#organization-member) | Regular active contributor in the community | Has pushed at least one PR to an Istio repository | Can recommend a contributor to become a member |
```
Comment (qiujian16) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
| [Organization Member](#organization-member) | Regular active contributor in the community | Has pushed at least one PR to an open-cluster-management-io repository | Can recommend a contributor to become a member |
```

File: CONTRIBUTOR_LADDER.md [hunk 2]
```diff
@@ -19,6 +19,13 @@ Each of the contributor roles below is organized into lists of three types of th
...
+| [Reviewer](#reviewer) | Hold expertise of a repo or a specific area | Need to review PRs | Can recommend and a member to become a reviewer |
+| [Maintainer](#maintainer) | Reviewing at least 20 PRs per year, especially PRs that involve multiple parts of the project, Participating in, and leading, community meetings | Experience as a Reviewer for at least 3 months, Can commit to spending at least 15 hours per month working on the project | Added to root OWNERs file in one or more Subprojects, Have a vote in Maintainer decision-making meetings and recommand a reviewer to become a maintainer |
```
Comment (qiujian16) [inline CONTRIBUTOR_LADDER.md]: we do not have this Maintainer decision-making meetings :)

File: CONTRIBUTOR_LADDER.md [hunk 3]
```diff
@@ -19,6 +19,13 @@ Each of the contributor roles below is organized into lists of three types of th
 
 Open Cluster Management (OCM) is a project made up of several Subprojects, such as Foundation and Policy.  This contributor ladder applies to each Subproject.  As such, a contributor advances in each Subproject separately, and may be a Member in one Subproject while being a Maintainer in another.  For the Open Cluster Management project overall, the contributor is considered to be the highest level they have reached in any Subproject.  Thus, someone who is a Project Member in one Subproject is a Project Member of OCM overall.
 
+| Role | Responsibilities | Requirements | Privileges |
+| ---- | ---------------- | ------------ | ---------- |
+| [Contributor](#contributor) | Casual contributor to the project | n/a | Invitations to contributor events, Eligible to become an Organization Member |
+| [Organization Member](#organization-member) | Regular active contributor in the community | Has pushed at least one PR to an Istio repository | Can recommend a contributor to become a member |
+| [Reviewer](#reviewer) | Hold expertise of a repo or a specific area | Need to review PRs | Can recommend and a member to become a reviewer |
```
Comment (qiujian16) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
| [Reviewer](#reviewer) | Hold expertise of a repo or a specific area | Need to review PRs | Can recommend a member to become a reviewer |
```

File: CONTRIBUTOR_LADDER.md (hunk 2 above)
Comment (qiujian16) [reply CONTRIBUTOR_LADDER.md]: root OWNERs is not correct, we have repos that owners are in the sub directory which means they are the maintainer of a certain subsystem.

File: CONTRIBUTOR_LADDER.md (hunk 1 above)
Comment (xuezhaojun) [reply CONTRIBUTOR_LADDER.md]: Updated.

File: CONTRIBUTOR_LADDER.md (hunk 3 above)
Comment (xuezhaojun) [reply CONTRIBUTOR_LADDER.md]: Update.

File: CONTRIBUTOR_LADDER.md (hunk 2 above)
Comment (xuezhaojun) [reply CONTRIBUTOR_LADDER.md]: Updated.

**Review by qiujian16 — COMMENTED** [review]
/approve
/assign @mikeshng

**Comment by lbocinco** [comment]
Don't forget to mention this contributor ladder also in the contributing guides, when it is updated.

**Comment by mikeshng** [comment]
> Don't forget to mention this contributor ladder also in the contributing guides, when it is updated.

Yes. Thank you @lenicatko for the reminder.

**Review by mikeshng — APPROVED** [review]
/lgtm

**Pull request merged**